### PR TITLE
Rename and use frontcontainer.tsx

### PR DESCRIPTION
**Pull request by Georges-GNM**
Co-authored-by: James Gorrie <james.gorrie@guardian.co.uk>

## What does this change?

Follow up to @mxdvl 's PR #7182 - this renames the component to FrontSection (as it's a section intended for fronts) and implements it in the fronts layout so that it respects the grid at various breakpoints. 

This also updates relevant stories (including all the currently supported container stories) to use the FrontSection component. A visual bug where the container title overflows into the container is resolved by adding the `overflow-wrap` property to the ContainerTitle component css; this would currently be occurring on frontend if the container title was a single, very long word. But to also match the stories' container titles more with the ones on PROD, the titles have been broken up (e.g. `FixedSmallSlowVMPU` becomes `Fixed Small Slow V MPU`)

## Why?

Fixes a [bug](https://github.com/guardian/dotcom-rendering/issues/6987) enabling us to get closer to launching fronts

## Screenshots

| Frontend   (largest breakpoint)   | DCR      |
|-------------|------------|
| ![frontendlargest][] | ![dcrlargest][] |

[frontendlargest]: https://user-images.githubusercontent.com/102960844/218741766-1941362c-ebfd-470c-925b-c8d665b7a49b.png
[dcrlargest]: https://user-images.githubusercontent.com/102960844/218741315-ca1eba67-1597-47ab-b09d-49cda0bf0759.png

| Frontend   (large breakpoint)   | DCR      |
|-------------|------------|
| ![Frontend (large)][] | ![DCR (large)][] |

[Frontend (large)]: https://user-images.githubusercontent.com/102960844/220395664-cf08bf68-a85e-4cf5-a2d5-3d12b354d7ef.png
[DCR (large)]: 
https://user-images.githubusercontent.com/102960844/220395188-74b9551d-f84d-4717-b654-316e236594fb.png

| Frontend   (small breakpoint)   | DCR      |
|-------------|------------|
| ![Frontend (small)][] | ![DCR (small)][] |

[Frontend (small)]: https://user-images.githubusercontent.com/102960844/220398613-6275007a-2637-4fe5-80e7-694403caa971.png

[DCR (small)]: https://user-images.githubusercontent.com/102960844/220398443-8a080ce2-88d9-411c-873f-f44611da0b14.png

### CODE REVIEW

**Comment by mxdvl** [comment]
The `aria-controls` need adjustment to get back to 98 on accessibility, annoyingly, the Lighthouse links no longer point to the actual report.

File: dotcom-rendering/src/web/components/ShowHideButton.tsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ const showHideButtonCss = (
 	text-decoration: none;
 
 	${from.wide} {
-		position: absolute;
+		position: relative;
```
Comment (Georges-GNM) [inline dotcom-rendering/src/web/components/ShowHideButton.tsx:25]: Absolute position moves the button out of view; relative position brings it back to where it's meant to be: 
image

image

File: dotcom-rendering/src/web/components/ContainerTitle.tsx [hunk 2]
```diff
@@ -57,10 +51,6 @@ const bottomMargin = css`
 
 const marginStyles = css`
 	margin-left: 0;
```
Comment (Georges-GNM) [inline dotcom-rendering/src/web/components/ContainerTitle.tsx:53]: This brings the title in line with frontend: 

Around the world

DCR before: 
Around the world

DCR now: 
image

File: dotcom-rendering/src/web/components/ShowHideButton.tsx (hunk 1 above)
Comment (mxdvl) [reply dotcom-rendering/src/web/components/ShowHideButton.tsx:25]: Looking at the original PR, I think [we could remove almost all of the styles](https://github.com/guardian/dotcom-rendering/commit/f22a22b4b2d910b7ed566b670fa7da8b0ebc6bbd#diff-d9a8d15e0990e7c3f8c2f36863a67ea118ce050b7d014757cbdda6ec80f1d251)

```tsx
```

**Comment by mxdvl** [comment]
There’s a real impact of using grid for performance, comparing [the Lighthouse report](https://storage.googleapis.com/lighthouse-infrastructure.appspot.com/reports/1676388859591-6203.report.html) with [our current baseline](https://storage.googleapis.com/lighthouse-infrastructure.appspot.com/reports/1676545286721-30692.report.html):

- DOM size reduced by 10% – 3,660 v 3,945
- Maximum DOM depth reduced by 15% – 19 v 22 
- Main thread work for style and layout is nearly halved with this PR – 781 ms v 1,444 ms

File: dotcom-rendering/src/web/components/Palettes.stories.tsx [hunk 3]
```diff
@@ -131,7 +131,7 @@ export const LongRunningPalette = () => (
 
 export const LongRunningAltPalette = () => (
 	<FrontSection
-		title="LongRunningAltPalette"
+		title="Long Running Alt Palette"
 		padContent={false}
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/Palettes.stories.tsx:135]: I think this should be removed!

File: dotcom-rendering/src/web/components/FrontSection.stories.tsx [hunk 4]
```diff
@@ -50,6 +50,9 @@ const Placeholder = ({
 	</div>
 );
 
+/** The stories below consistently have padContent set to false, even though the default value and the one which ends up rendering the component in the application is true.
+ * This is so the stories below can match up with the section stories which the FrontSection component is modeled after. */
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/FrontSection.stories.tsx:54]: I fear this leads to having stories which look different to what is in production, which is of very little value.

Comment (Georges-GNM) [reply dotcom-rendering/src/web/components/FrontSection.stories.tsx:54]: I agree; I wrote that reasoning at first that these stories wouldn't directly translate into prod, and the ones that do (e.g. the containers with cards in them) were working. But it's still not a satisfying solution, so been trying to find one!

Comment (mxdvl) [reply dotcom-rendering/src/web/components/FrontSection.stories.tsx:54]: I would probably stick with what I did in the original PR and make the stories look as much like PROD as possible and remove `padContent` entirely.

**Review by jamesgorrie — COMMENTED** [review]
This is great work, really thorough and well tested. Quite a large PR, but that's the nature of refactoring components that are quite high up the chain.

**Review by jamesgorrie — APPROVED** [review]

File: dotcom-rendering/src/web/components/ContainerTitle.tsx [hunk 5]
```diff
@@ -35,7 +35,8 @@ const headerStyles = (fontColour?: string) => css`
 	${headline.xsmall({ fontWeight: 'bold' })};
 	color: ${fontColour ?? neutral[7]};
 	padding-bottom: ${space[1]}px;
-	padding-top: ${space[1]}px;
+	padding-top: 6px;
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/ContainerTitle.tsx:38]: I’m not sure why we would move away from `@guardian/source-foundation`’s space, here

File: dotcom-rendering/src/web/components/ShowHideButton.tsx (hunk 1 above)
Comment (mxdvl) [reply dotcom-rendering/src/web/components/ShowHideButton.tsx:25]: I still think we can remove a lot more styles!

**Review by mxdvl — APPROVED** [review]
Excellent diligence, here! I would _highly_ encourage you to squash your commits, either before merging or via the “Squash & Merge” command to make it easier for future travellers. Happy to pair.

Two small points unadressed:
- why go for a pixel-defined space rather than one based on Source?
- why keep all of the unused button styling?

File: dotcom-rendering/src/web/components/ContainerTitle.tsx (hunk 5 above)
Comment (Georges-GNM) [reply dotcom-rendering/src/web/components/ContainerTitle.tsx:38]: Thought I'd commented on that before - comparing the padding-top value that's currently on frontend to the one that gets generated with source, the available source values don't seem to match. space[1] I believe is 4, space[2] is 8.

**Comment by Georges-GNM** [comment]
> Excellent diligence, here! 🎉 📐
> 
> I would encourage you to squash your commits, either before merging or via the “Squash & Merge” command to make it easier for future travellers. Happy to pair.
> 
> Two small points unadressed:
> 
> * why go for a pixel-defined space rather than one based on Source?
> * why keep all of the unused button styling?

Thanks! Have removed the button styling and squashed and merged 👍

File: dotcom-rendering/src/web/components/FrontSection.tsx [hunk 6]
```diff
@@ -520,23 +527,24 @@ export const FrontSection = ({
 			{isToggleable && (
 				<div css={sectionShowHide}>
 					<ShowHideButton
-						sectionId={sectionId}
+						sectionId={`container-${sectionId}`}
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/FrontSection.tsx]: Is this going to work?

File: dotcom-rendering/src/web/components/FrontSection.tsx [hunk 7]
```diff
@@ -24,8 +24,6 @@ type Props = {
 	centralBorder?: 'partial' | 'full';
```
Comment (mxdvl) [inline dotcom-rendering/src/web/components/FrontSection.tsx:24]: I actually think that for `FrontContainer`, the central border should always be `full`.

**Pull request merged**